File: src/main/java/net/hectus/neobb/turn/default_game/mob/TBlaze.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.function.AttackFunction;
import org.bukkit.entity.Blaze;

public class TBlaze extends MobTurn<Blaze> implements AttackFunction, HotClazz {
    public TBlaze(NeoPlayer player) { super(player); }
    public TBlaze(Blaze data, NeoPlayer player) { super(data, player); }

    @Override
    public void apply() {
        player.game.addModifier("warp-prevent.cold");
        player.game.addModifier("warp-prevent.water");
    }

    @Override
    public int cost() {
        return 5;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/mob/TBlaze.kt
package net.hectus.neobb.modes.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.Modifiers
import org.bukkit.entity.Blaze

class TBlaze(data: Blaze?, cord: Cord?, player: NeoPlayer?) : MobTurn<Blaze>(data, cord, player), AttackFunction, HotClazz {
    override val cost: Int = 5

    override fun apply() {
        player!!.game.addModifier(Modifiers.Game.NO_WARP.name + "_cold")
        player.game.addModifier(Modifiers.Game.NO_WARP.name + "_water")
    }
}
